Cython/Plex/Scanners.py

```diff
@@ -61,10 +63,14 @@ class Scanner(object):
     #  start_pos = 0         # position in input of start of token
     #  start_line = 0        # line number of start of token
     #  start_col = 0         # position in line of start of token
+    #  These positions are used to track what was read from the queue
+    #   (which may differ from the internal state when tokens are replaced onto the queue)
+    #  last_token_line       # line number of the last token taken from the queue
+    #  last_token_col        # position in line of the last token taken from the queue
```

Comment:
Looks like these could just be stored as a tuple. Looking at the usage, maybe not even a ctuple but a plain Python tuple.

However, why wasn't it an option to reset `start_line` and `start_col` instead of adding new attributes?

Reply:
I've combined the two comments:
* got rid of `last_token_line`, `last_token_col`, `start_line` and `start_col`
* converted it to a tuple (with all three position elements in it - although the filename doesn't change it seems easier to make it part of the tuple)

Reply:
Changed my mind slightly - I think we do have to store a separate position of "position that the scanner read from" to get good diagnostics when actions go wrong. I've stored that as a tuple as well since it isn't really used for anything except producing error messages.